cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -129,23 +154,21 @@ fi
 
 
 function configure_sql_client(){
-  # Configure mysql client to talk to metastore as root.
+  # Configure mysql client to talk to metastore
   cat << EOF > /etc/mysql/conf.d/cloud-sql-proxy.cnf
 [client]
 protocol = tcp
 port = ${metastore_proxy_port}
-user = root
```

Comment:
This is technically a breaking change, right? If you aren't providing the user/password here, it falls back to the GCE service account. https://cloud.google.com/sql/docs/mysql/sql-proxy#invocation

Two questions:

1) If you're falling back to the GCE service account, assuming it has permission for Cloud SQL, what user does it authenticate with to MySQL?

2) Would this init action previously have worked in a cross-project setup and would this change break that? (That's not necessarily an issue, just curious).

Maybe we can authenticate as `root` if a) the user specifies the root pasword, or b) if the user specifies neither the hive nor root password. That way we avoid a breaking change.

Reply:
The patch should not break legacy code. The `[client]` section in `cloud-sql-proxy.cnf` is to provide default parameters for the `mysql` client (see [doc](https://dev.mysql.com/doc/refman/8.0/en/option-files.html)). What I did in my patch is explicitly provide a user and password every time the `mysql` client is used. This removes the need to provide the `root` user (in the case where the database and `hive` user already exist).

To answer your questions:

1. So, technically there's no "falling back" to a service account. The service account is only used to establish the proxy connection with the Cloud SQL instance (i.e. the underlying VM). The `root` and `hive` user are completely independent from the service account. Those users are only used by the `mysql` client to run queries on the MySQL server.

2. I believe both the old and new versions of this init action would work across projects. That's why you have to provide the project ID in the `hive-metastore-instance` metadata key. If the service account is given the `cloudsql.client` role in the project where the Cloud SQL instance is hosted, then it would be allowed to establish a proxy connection with it.